visma/matrix/structure.py

```diff
@@ -102,6 +103,24 @@ class SquareMat(Matrix):
     def determinant(self):
         pass
 
+    def traceMat(self):
+        """Returns the trace of a square matrix (sum of diagonal elements)
+
+        Arguments:
+            mat {visma.matrix.structure.Matrix} -- matrix token
+
+        Returns:
+            trace {visma.matrix.structure.Matrix} -- string token
+        """
+        from visma.simplify.simplify import simplify
+        trace = []
+        for i in range(self.dim[0]):
+            trace.extend(self.value[i][i])
+            trace.append(Binary('+'))
+        trace.append(Constant(0))
+        trace, _, _, _, _ = simplify(trace)
+        return trace
+
```

Comment:
and change this function as follows:
`

        from visma.matrix.operations import simplifyMatrix
        trace = Matrix()
        trace.empty([1,1])
        for i in range(self.dim[0]):
            trace.value[0][0].extend(self.value[i][i])
            trace.value[0][0].append(Binary('+'))
        trace.value[0][0].append(Constant(0))
        trace = simplifyMatrix(trace)
        return trace

`

Reply:
This is the function that I had written initially. But this returns trace as a 1x1 matrix element. There was no need for it, and hence the code was changed to return a string token. The function works perfectly.